### DISCUSSION TITLE
Embed necessary parts of microG directly into non-root reVanced ?

### DISCUSSION THREAD

**Discussion by ghost** [ReVanced/revanced-discussions#483] [Question] [answered]
Hi, sorry for bad English.

Is it possible to take the necessary microG components, such as the account login mechanism, McsService, cast, etc., and embed it in a non-root reVanced as a patch ? Non-root Youtube ReVanced worning without microG as standalone app... sounds like a dream ?

If it's possible to implement, you will make thousands of people happy.

**Comment by oSumAtrIX** [comment, marked as answer]
Yes it is technically possible but practically the same. Simply install MicroG and you are gtg.

**Comment by SaurabhKaperwan** [comment]
Micro g drains battery

**Reply by oSumAtrIX** [reply]
Simply moving the same code to YouTube is not any different.
